src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -175,6 +182,34 @@ public boolean canBeGathered()
         return toKeep;
     }
 
+    // Adding request system aware versions of the furnace recipes, to make keeping stock easier, and allow crafting chains. 
+    // To find what to add, we'll run through the predefined ingredients, and if any of them are smeltable, we'll add the recipe
+    // Something to consider in the future is adding the concept of 'free' recipes that don't count against the building limit. 
+    @Override
+    public void checkForWorkerSpecificRecipes()
+    {
+
+        ResourceLocation ingredients = new ResourceLocation("minecolonies", this.getJobName().toLowerCase().concat("_ingredient"));
+
+        for (final Item item : ItemTags.getCollection().getOrCreate(ingredients).getAllElements())
+        {
+            final ItemStack smeltingResult = FurnaceRecipes.getInstance().getSmeltingResult(new ItemStack(item, 1));
+            if (ItemStackUtils.CAN_EAT.test(smeltingResult))
+            {
+
+                final IRecipeStorage storage = StandardFactoryController.getInstance().getNewInstance(
+                    TypeConstants.RECIPE,
+                    StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN),
+                    ImmutableList.of(new ItemStack(item, 1)),
+                    1,
+                    smeltingResult,
+                    Blocks.FURNACE);
+          
+                addRecipeToList(IColonyManager.getInstance().getRecipeManager().checkOrAddRecipe(storage));
+            }
+        }
+    }
+    
```

Comment:
I'm nt a fan at all of this. Adding a lot of automatic recipes slows down the colony (each time a request is executed all recipes have to be checked). Why do we need this explicitly?

Reply:
In a vanilla world, we really don't. Once you add in modpacks, there are a lot of recipes that require cooked foods that can't be fulfilled today. As we discussed in Discord, I'll look at the temporary recipe via resolver.